File: 10th/aof_2024_10/Solver.java
package aof_2024_10;

public class Solver {
    public static void main(String[] args) {
        //FileParser fileParser = new FileParser("test.txt");
        FileParser fileParser = new FileParser("input.txt");
        Map map = new Map(fileParser.readFile());
        PathFinder pathFinder = new PathFinder(map);

        System.out.println("Advent of Code 2024 - Day 10");
        System.out.println("Part 1: " + pathFinder.findAllPathsFor1());
        System.out.println("Part 2: " + pathFinder.findAllPathsFor2());
    }
}

File: 11th/solver.py
import os
from functools import cache
#pylint: disable=missing-function-docstring

def file_reader(file_path):
    with open(os.path.join(os.path.dirname(__file__), file_path), 'r') as f:
        return f.read().splitlines()

def parser(lines):
    return lines[0].strip().split(" ")

def solver(puzzle_input):
    return sum(blink_recurse(int(n), 75) for n in puzzle_input)

def blink(number):
    if number == 0:
        return [1]
    str_nr = str(number)
    len_nr = len(str_nr)
    if len(str_nr) % 2 == 0:
        return [int(str_nr[:len_nr//2]), int(str_nr[len_nr//2:])]
    return [number * 2024]

@cache # that was rediculously effective
def blink_recurse(number, blinks):
    if blinks == 0:
        return 1
    newNumbers = blink(number)
    return sum(blink_recurse(n, blinks - 1) for n in newNumbers)


FILE = "input.txt"

INPUT = parser(file_reader(FILE))

print("Advent of Code: Day 11")
print("Solution for Part 2:", solver(INPUT))
